public.tf

```diff
@@ -13,7 +20,11 @@ resource "aws_subnet" "public" {
   vpc_id            = "${data.aws_vpc.default.id}"
   availability_zone = "${element(var.availability_zones, count.index)}"
   cidr_block        = "${cidrsubnet(signum(length(var.cidr_block)) == 1 ? var.cidr_block : data.aws_vpc.default.cidr_block, ceil(log(length(data.aws_availability_zones.available.names) * 2, 2)), length(data.aws_availability_zones.available.names) + count.index)}"
-  tags              = "${module.public_label.tags}"
+  tags              = {
+    "Name"          = "${module.public_subnet_label.id}-${element(var.availability_zones, count.index)}"
```

Comment:
Let's replace `-` with `var.delimiter` and `-` sign occurrences in `element(var.availability_zones, count.index)` to `var.delimiter`.

Reply:
wow this was a lot trickier than expected